base/linalg/diagonal.jl

```diff
@@ -101,9 +101,9 @@ end
 parent(D::Diagonal) = D.diag
 
 ishermitian{T<:Real}(D::Diagonal{T}) = true
-ishermitian(D::Diagonal) = all(D.diag .== real(D.diag))
+ishermitian(D::Diagonal) = isreal(D.diag)
```

Comment:
Regardless of how useful it is, since we define a vector-space algebra for arrays of numbers, the `zero` and related functions need to be defined on arrays for mathematical completeness.